fiat-core/src/main/java/com/netflix/spinnaker/fiat/model/UserPermission.java

```diff
@@ -38,6 +38,20 @@
   private Set<Resource> extensionResources = new LinkedHashSet<>();
   private boolean admin = false;
 
+  /**
+   * Custom setter to normalize the input. lombok.accessors.chain = true is enabled, so setters must
+   * return {@code this}.
+   */
+  public UserPermission setId(String id) {
+    this.id = id.toLowerCase();
+    return this;
+  }
+
+  /** Custom getter to normalize the output. */
+  public String getId() {
+    return this.id.toLowerCase();
+  }
+
```

Comment:
Maybe a bit hacky?  The alternative would be scattering normalize code throughout the permission resolver and repository (and probably other places too).

Reply:
I think this makes sense and is nicely localized

Should we do something at the API to ensure we lowercase request parameters before we start calling things like permission repo? I think a number of those paths just go by userId

Reply:
nevermind .. I see ControllerSupport below!